src/shared/telemetry/spans.ts

```diff
@@ -183,33 +200,39 @@ export class TelemetryTracer extends TelemetryBase {
      * on existing spans has no effect on the active span.
      */
     public get activeSpan(): TelemetrySpan | undefined {
-        return this.#context.getStore()?.activeSpan
+        return this.#context.getStore()?.spans[0]
+    }
+
+    /**
+     * State that is applied to all new spans within the current or subsequent executions.
+     */
+    public get attributes(): Readonly<Attributes> {
+        return this.#context.getStore()?.attributes ?? this.#syncAttributes
     }
 
     /**
-     * Records information on all _current_ spans in the current execution context.
+     * Records information on all current and future spans in the execution context.
```

Comment:
Does it apply to all future spans or only the _next_ span/context (which "pops it off the stack")?

Reply:
All future spans within the current and any nested contexts. 

The only exception is when you're not in a context. In this case it only affects the next span/context. But I think I'll remove this behavior and require that the caller be within a context.